### PR TITLE
refactor(messaging): route provider preparation through adapter

### PR DESCRIPTION
**Pull request by rsliter**
## Outcome

Messaging onboarding now routes checked-in profile inspection and import, provider lookup and update, and post-update verification through typed `OpenShellProviderAdapter` results. Named gateway targets fail closed when an ambient `OPENSHELL_GATEWAY_ENDPOINT` could redirect the operation, while lookup, update, identity-mismatch, and operational failures retain distinct redacted diagnostics.

## Reason

The accepted #9806 slice requires the adapter to own the OpenShell CLI boundary instead of leaving profile and provider lifecycle commands in onboarding. This is a fresh implementation from current `main`; it does not merge or cherry-pick the closed #10719 attempt.

### Related issues

- Part of #9806
- Replaces the adapter foundation attempted in #10719
- Relates to #9813, which owns migration of the remaining raw CLI consumers
- Provides a fresh base on which the separate #10724 and #10726 consumer slices can be restacked

## Changes

- Add typed provider metadata, lookup, update, profile-import, and verification results to `OpenShellProviderAdapter`.
- Move checked-in profile parsing, validation, import, export, and exact contract verification behind the CLI adapter.
- Move provider metadata and diagnostic parsing to the adapter layer, retaining only the narrow exact-not-found classification needed at the CLI boundary.
- Route messaging provider publication and managed-clone reconciliation through exact adapter calls.
- Preserve safe redacted lookup and update failure details while keeping identity mismatch distinct from operational failure.
- Preserve both provider and temporary-source cleanup failures when preparation aborts.
- Reject named-target operations when ambient `OPENSHELL_GATEWAY_ENDPOINT` is present.
- Bind both ordinary and Hermes portable sandbox creation to the same provider-effect boundary, including deferred post-identity effects and resume replay protection.
- Add tests for exact `getProvider`, `updateProvider`, profile-import, and verification calls, including raw-command exclusion and repeatable desired-state recovery after partial publication.

## Verification

- On exact candidate tree `7e3490e69`, the focused CLI suite passed 253 tests across seven files, covering adapter get/update/profile results, provider publication, ordinary creation, Hermes portable creation, and the real superseded portable transaction. The published signed candidate `364aa89d4` has that exact tree.
- `npx vitest run --project integration test/onboarding/onboard-hermes-portable-provider-publication.test.ts`: four public-boundary cases passed on `364aa89d4`, covering pre-create publication, deferred post-verification publication, verified-resume suppression, superseded-path exclusion, exact named-gateway adapter calls, and isolation from ambient XDG paths.
- `npm run typecheck:cli`: passed on `364aa89d4` after the final canonical `main` refresh.
- Targeted Oxlint for the changed TypeScript files: passed.
- `npm run validate:pr`: passed on exact signed candidate `364aa89d4` against canonical `main` `d836ccb44` in a clean isolated checkout, including repository checks, secret scanning, source-shape checks, growth guardrails, commit policy, and the CLI pre-push type check.
- The full manual-stage coverage pass completed successfully. The broader all-files pre-commit sweep passed every check except existing Hadolint warnings in unchanged Dockerfiles; the identical Hadolint failure was reproduced on canonical `main` `3509b5a43` before the subsequent `main` refresh.
- `git diff --check origin/main...HEAD`: passed.
- `npm run review:local`: previously failed before analysis because its desktop bootstrap did not forward the active Colima `DOCKER_HOST`; no artifacts were produced. Per maintainer direction, repairing that unrelated local-review tooling is deferred.
- PR Review Advisor: three full runs on `eb2be67fd` produced no specialist artifacts and were deferred as unavailable infrastructure evidence. On final head `c7217b54b`, run `33759103328` succeeded for all nine specialists. Every specialist summary and full JSONL session was read; no specialist reported a change-required finding.
- Diff inspection: no secrets, API keys, or credentials are present.

## Review notes

- CodeRabbit's final incremental review covered `eb2be67fd..c7217b54b` and produced no actionable comments. Its generated summary retained an older Hermes portable risk sentence, but CodeRabbit rechecked commits `1abb93872` and `49bc39aef`, verified all four portable lifecycle cases, and explicitly classified that sentence as stale. All review threads are resolved.
- CodeRabbit's docstring-coverage warning is a generic advisory, not a repository gate. Adding docstrings across 51 touched existing functions would broaden this migration without protecting the requested adapter behavior, so no change is required.
- The remaining raw provider consumers are intentionally unchanged for #9813.
- The diagnostic parser could not be removed entirely because the CLI adapter still needs exact absence classification. It is no longer an onboarding shim and now lives at the adapter boundary.
- The branch includes canonical `main` through `d836ccb44`, including the remediation for the inherited `fast-uri` advisories, the package-contract npm fix from #10986, the stricter existing-profile validation protocol from #10884, and the Advisor on-demand diff-reading fix from #10952.
- #10724 and #10726 are broader drafts stacked on the closed #10719 branch. This PR does not close or modify them.

---
Signed-off-by: Rebecca Sliter <571084+rsliter@users.noreply.github.com>

### CODE REVIEW

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

File: src/lib/onboard/sandbox-create/provider-publication.test.ts [hunk 1]
```diff
@@ -108,12 +70,29 @@ function publicationInput(
...
+
+async function prepareProviders(
   input: Parameters<typeof publishAttachedProvidersBeforeDockerSandboxCreation>[0],
   deps: Parameters<typeof publishAttachedProvidersBeforeDockerSandboxCreation>[1],
-): void {
-  validateAttachedMessagingProvidersBeforeSandboxCreation(input, deps);
-  publishAttachedProvidersBeforeDockerSandboxCreation(input, deps);
+): Promise<void> {
+  await validateAttachedMessagingProvidersBeforeSandboxCreation(input, deps);
+  await publishAttachedProvidersBeforeDockerSandboxCreation(input, deps);
```
[the remark this answers is not in the document]
Comment (rsliter) [reply src/lib/onboard/sandbox-create/provider-publication.test.ts:93]: Verified against the current test boundaries. `provider-publication.test.ts` covers the changed adapter boundary with exact `getProvider`, `updateProvider`, profile-import, and verification calls, and makes any raw provider command fail. Existing orchestration and `handleSandboxState` tests own deferred, fresh, and resumed lifecycle staging. Driving the full onboarding entrypoint through the same four-way matrix would require a new test-only adapter injection seam and duplicate lifecycle behavior this slice does not change. The valid partial-publication risk is now covered at `afaa515b8` by a desired-state rerun test that asserts all four `updateProvider` calls and excludes the raw command path.

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
Advisor artifact follow-up for d0f93d667: the code-reduction finding says `prepareProviders` has one caller. It has two callers at provider-publication.test.ts:100 and :500. The second exercises binding drift across the validation and publication phases. Removing the helper made that test fail before its assertion, so I restored the unchanged helper. Focused result after restoration: 18/18 passed; CLI type checking passed. No code change is warranted for this finding.

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai review

**Comment by rsliter** [comment]
@coderabbitai The remaining merge-risk sentence about Hermes portable provider effects is stale on the current head. Commit 1abb938 routes both pre-create and deferred effects through the portable path; test/onboarding/onboard-hermes-portable-provider-publication.test.ts covers non-deferred ordering, deferred ordering, verified-resume suppression, and superseded exclusion with exact adapter calls; and 49bc39a exercises the real portable transaction twice and proves the superseded pass never enters the provider-owned create callback. Please verify this evidence and clear or update that risk note.

**Comment by rsliter** [comment]
Review note: the required `build-typecheck` and aggregate `checks` failures are inherited from canonical `main`, not introduced by this PR.

Both jobs fail in `test/package-contract/managed-image-registry-transport.test.ts` while npm 10.9.8 runs the packed CLI `--omit=dev` install, with `Cannot read properties of null (reading 'edgesOut')`:

- [PR job](https://github.com/NVIDIA/NemoClaw/actions/runs/33759107286/job/100667124030)
- [latest `main` job at `f7c1d8354`](https://github.com/NVIDIA/NemoClaw/actions/runs/33782279441/job/100738919758)

This PR does not change that package-contract test, the root package manifest, or the lockfile. The latest-head Advisor run completed successfully across all nine specialists, CodeRabbit is green, and all review threads are resolved. The inherited required check must still clear before merge.

**Review by prekshivyas — APPROVED** [review]

**Review by prekshivyas — DISMISSED** [review]
Requesting changes pending completion of validation:

I did not identify a separate code-level defect in the current head. However, a fresh full CI, image-build, and security validation run is currently in progress. Please allow that run to complete successfully and address any failures before this PR is merged. This request is validation-only and can be cleared once the current head has complete passing evidence.

No merge conflict was detected against the latest main.

**Review by apurvvkumaria — APPROVED** [review]
Reviewed commit 364aa89d4. No blocking correctness or security finding.\n\nPR Advisor feedback is addressed in the current code and tests. The latest exact-head run completed five specialists with no required change; four specialist jobs failed before analysis on repeated HTTP 429 responses. The prior complete nine-specialist run reviewed the unchanged implementation patch and reported no required change. All three CodeRabbit threads are resolved.\n\nSecurity review passed all nine rubric categories. Credential values remain in child environments rather than arguments; named gateway targeting rejects ambient endpoint authority; metadata parsing is bounded and fails closed; cleanup and recovery preserve causal failures; no dependency, workflow, policy, or image surface changed.\n\nValidation: 219 focused source tests passed, the new Hermes portable integration and fixture-contract tests passed, CLI type-checking passed, and the diff check passed. The local onboard-messaging gateway-authority failure reproduces unchanged on the base commit. Current-head CI has 48 passing checks; the only failures are the four Advisor rate-limit failures described above.
